network/tcp_test.go

```diff
@@ -373,13 +373,71 @@ func TestTCPConnWithListener(t *testing.T) {
 	}
 }
 
-// will create a TCPListener & open a golang net.TCPConn to it
+// will create a TCPListener with a specific address to listen on &
+// open a golang net.TCPConn to it
+func TestTCPListenerWithListenAddr(t *testing.T) {
+	// Ensuring that we cannot only have the port as a listen address.
+	addr := NewAddress(PlainTCP, "1.2.3.4:0")
+	listenAddr := ":0"
+	_, err := NewTCPListenerWithListenAddr(addr, tSuite, listenAddr)
+	require.NotNil(t, err, "Should fail: cannot only have a port as a listen address")
+
+	// Testing different working configurations.
+	addrs := [2]Address{NewAddress(PlainTCP, "1.2.3.4:0"), NewAddress(PlainTCP, "1.2.3.4:0")}
+	listenAddrs := [2]string{"127.0.0.1", "127.0.0.1:0"}
+	expectedHosts := [2]string{"127.0.0.1", "127.0.0.1"}
+	for i := 0; i < len(addrs); i++ {
```

Comment:
Something like this is more go-like:
```
testVector := struct{addr Address; listen string; expected string}{{NewAddress(...), "127.0.0.1", "127.0.0.1"}, {}}
for _, tv := range testVector{
  ln, err := NewTCPListenerWithListenAddr(tv.addr, tSuite, tv.listen)
...
```

see also: https://github.com/dedis/onet/blob/cd17dbbe5846e58a8194c1bc63ebed142cbed454/network/address_test.go#L51-L85